serial/tools/list_ports_linux.py

```diff
@@ -90,20 +90,22 @@ def read_line(self, *args):
 
 def comports(include_links=False, hide_subsystems=["platform"]):
     devices = set()
-    for driver in open('/proc/tty/drivers'):
-        items = driver.readlines().strip().split()
-        try:
-            if items[4] == 'serial':
-                devices.update(glob.glob(items[1]+'*'))
-        except IndexError:
-            devices.update(glob.glob('/dev/ttyS*'))     # built-in serial ports
-            devices.update(glob.glob('/dev/ttyUSB*'))   # usb-serial with own driver
-            devices.update(glob.glob('/dev/ttyXRUSB*')) # xr-usb-serial port exar (DELL Edge 3001)
-            devices.update(glob.glob('/dev/ttyACM*'))   # usb-serial with CDC-ACM profile
-            devices.update(glob.glob('/dev/ttyAMA*'))   # ARM internal port (raspi)
-            devices.update(glob.glob('/dev/rfcomm*'))   # BT serial devices
-            devices.update(glob.glob('/dev/ttyAP*'))    # Advantech multi-port serial controllers
-            devices.update(glob.glob('/dev/ttyGS*'))    # https://www.kernel.org/doc/Documentation/usb/gadget_serial.txt
+    with open('/proc/devices') as file_handler:
```

Comment:
What do you expect to find in /proc/devices ?

Reply:
Again, brain fart. Fixed it as soon as I actually tested the code. My bad.